airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryDestinationTest.java

```diff
@@ -374,11 +322,17 @@ void testWriteFailure(final DatasetIdResetter resetDatasetId) throws Exception {
   }
 
   private Set<String> fetchNamesOfTablesInDb() throws InterruptedException {
+    if(dataset == null || bigquery == null) {
```

Comment:
I think the two conditionals on lines 325-327 and 333-335 can be combined. The `QueryJobConfiguration` AFAIK is just building a query rather than executing it. Still a little rusty on my Java but I think an `Optional` would be good here

```
Optional<Dataset> potentialDataset = Optional.ofNullable(dataset);
if (potentialDataset.filter(d -> !d.exists()).orElse(false) || bigquery == null) { 
  return Collections.emptySet();
}
  
```